photon-api/src/main/scala/com/linkedin/photon/ml/model/RandomEffectModel.scala

```diff
@@ -120,16 +149,41 @@ protected[ml] class RandomEffectModel(
   }
 
   // TODO: Violation of the hashCode() contract
+  /**
+   *
+   * @return
+   */
   override def hashCode(): Int = super.hashCode()
 }
 
 object RandomEffectModel {
 
   /**
-   * Compute the score for the dataset
+   * Determine the random effect model type: even though the model has many sub-problems, there is only one loss
+   * function type for a given random effect model.
```

Comment:
This is basically the `checkInvariants` method you had before - it's checking that we don't load in multiple GLMs of differing types as part of one random effect model.